internal/service/thirdpartyintegration/data_source_third_party_integrations.go

```diff
@@ -65,26 +65,24 @@ func flattenIntegrations(d *schema.ResourceData, integrations *matlas.ThirdParty
 
 func integrationToSchema(d *schema.ResourceData, integration *matlas.ThirdPartyIntegration) map[string]any {
 	integrationSchema := schemaToIntegration(d)
+
 	if integrationSchema.APIKey == "" {
 		integrationSchema.APIKey = integration.APIKey
 	}
 	if integrationSchema.ServiceKey == "" {
-		integrationSchema.APIKey = integration.ServiceKey
+		integrationSchema.ServiceKey = integration.ServiceKey
 	}
 	if integrationSchema.APIToken == "" {
-		integrationSchema.APIKey = integration.APIToken
+		integrationSchema.APIToken = integration.APIToken
 	}
 	if integrationSchema.RoutingKey == "" {
-		integrationSchema.APIKey = integration.RoutingKey
+		integrationSchema.RoutingKey = integration.RoutingKey
 	}
 	if integrationSchema.Secret == "" {
-		integrationSchema.APIKey = integration.Secret
+		integrationSchema.Secret = integration.Secret
 	}
 	if integrationSchema.Password == "" {
-		integrationSchema.APIKey = integration.Password
-	}
-	if integrationSchema.UserName == "" {
-		integrationSchema.APIKey = integration.UserName
+		integrationSchema.Password = integration.Password
```

Comment:
This was clearly incorrect, I believe it is being used for the case of the data source where if the tf config has no value, it will define the obfuscated values coming from the api. Removed username as it is always returned by the API.